.github/workflows/bench.yml

```diff
@@ -1,52 +1,135 @@
+                echo "Error: Invalid service name '${{ matrix.service }}'"
+                exit 1
+            fi
+            if ! ./run_benchmarks.sh ${{ matrix.service }}; then
+                echo "Error: Benchmark failed for ${{ matrix.service }}"
+                exit 1
+            fi
+            echo "Benchmark completed for ${{ matrix.service }}"
+
+
+      - name: List benchmark files
+        run: |
+          echo "Contents of current directory:"
+          ls -la
+          echo "Matching bench*.txt files:"
+          ls -la bench*.txt || echo "No matching files found"
+
+          
+      - name: Upload benchmark results
+        uses: actions/upload-artifact@v3
+        with:
+          name: benchmark-results
+          path: bench*.txt
+
+
+  analyze:
+    needs: benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout (GitHub)
+        uses: actions/checkout@v4
+      
+      - name: Download all benchmark results
+        uses: actions/download-artifact@v3
+        with:
+          name: benchmark-results
+          path: .
+      
+      - name: List downloaded artifacts
+        run: ls -la bench*.txt || echo "No matching files found"
+      
+      - name: output of tailcall
+        run: |
+          echo "Benchmark 1"
+          cat ./bench1_result1_graphql_tailcall_run.sh.txt
+          cat ./bench1_result2_graphql_tailcall_run.sh.txt
+          cat ./bench1_result3_graphql_tailcall_run.sh.txt
+          echo "End of Becnhmark 1"
+          echo ""
+          echo "Benchmark 2"
+          cat ./bench2_result1_graphql_tailcall_run.sh.txt
+          cat ./bench2_result2_graphql_tailcall_run.sh.txt
+          cat ./bench2_result3_graphql_tailcall_run.sh.txt
+          echo "End of Becnhmark 2"
+          echo ""
+          echo "Becnhmark 3"
+          cat ./bench3_result1_graphql_tailcall_run.sh.txt
+          cat ./bench3_result2_graphql_tailcall_run.sh.txt
+          cat ./bench3_result3_graphql_tailcall_run.sh.txt
+          echo "End of Benchmark 2"
+          echo ""
+
+      - name: Analyze results
+        run: |
+          rm results.md
+          for bench in 1 2 3; do
+            ls bench${bench}*.txt || echo "No matching files found for bench${bench}*.txt"
+            echo "Processing files for bench${bench}:"
+            bash analyze.sh bench${bench}*.txt
+            echo bench${bench}*.txt
+          done
+      
       - name: Print benchmark results
         run: cat ./results.md
-
+      
       - name: Comment benchmark results on PR
         if: github.event_name == 'pull_request_target'
         uses: peter-evans/commit-comment@v3
         with:
           sha: ${{ github.event.pull_request.head.sha }}
           body-path: "results.md"
           reactions: eyes
-
+      
       - name: Commit and push changes (on main branch)
-        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        if: github.ref == 'refs/heads/main'
```

Comment:
Change the commit_message to `[ci skip] update performance results in README.md` if you are removing the push event.